### PR TITLE
Dag info

### PR DESCRIPTION
**Pull request by zhujiming**
add dag info @junjiejiangjjj @filip-halt

### CODE REVIEW

File: towhee/functional/mixins/dataset.py [hunk 1]
```diff
@@ -25,6 +26,7 @@ class DatasetMixin:
 
     # pylint: disable=import-outside-toplevel
     @classmethod
+    @register_dag
```
Comment (reiase) [inline towhee/functional/mixins/dataset.py:29]: we don't need to register a data source as DAG

**Review by reiase — CHANGES_REQUESTED** [review]

File: towhee/functional/data_collection.py [hunk 2]
```diff
@@ -447,32 +467,26 @@ def range(*arg, **kws):
...
-
         Args:
             size (int): window size;
             drop_tail (bool): drop tailing windows that not full, defaults to False;
             raw (bool): whether to return raw data instead of DataCollection, defaults to True
-
         Returns:
             DataCollection of batched windows or batch raw data
-
         Examples:
```
[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/data_collection.py]: fixed

[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/data_collection.py]: fixed

File: towhee/functional/mixins/dataset.py (hunk 1 above)
Comment (zhujiming) [reply towhee/functional/mixins/dataset.py]: fixed

Comment (zhujiming) [reply towhee/functional/mixins/dataset.py]: fixed

File: towhee/functional/mixins/dag.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            else:
+                self.child_ids = [x.id for x in children]
+            info = {'op': self.op,
+                    'op_name': self.op_name,
+                    'is_stream': self.is_stream,
+                    'init_args': self.init_args,
+                    'call_args': self.call_args,
+                    'parent_ids': self.parent_ids,
+                    'child_ids':  self.child_ids}
+            self.control_plane.dag[self.id] = info
```
[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/mixins/dag.py]: fixed

[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/mixins/dag.py]: fixed

File: towhee/functional/data_collection.py [hunk 4]
```diff
@@ -872,7 +894,8 @@ def _factory(self, iterable, parent_stream=True, mode=None):
         with param_scope() as hp:
             hp().data_collection.parent = self
             df = DataFrame(iterable)
-            df._mode = self._mode if mode is None else mode
+            if mode is not None:
```
[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/data_collection.py]: fixed

File: towhee/functional/data_collection.py [hunk 5]
```diff
@@ -334,37 +345,39 @@ def map(self, *arg):
         >>> dc.map(lambda x: x+1).map(lambda x: x*2).to_list()
         [4, 6, 8, 10]
         """
+        unary_op = arg[0]
```
[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/data_collection.py:340]: just code style

File: towhee/functional/data_collection.py [hunk 6]
```diff
@@ -408,13 +426,14 @@ def inner(x):
             return unary_op(x)
 
         if hasattr(self._iterable, 'filter'):
-            return self._factory(self._iterable.filter(unary_op))
-
-        if hasattr(self._iterable, 'apply') and hasattr(unary_op, '__dataframe_filter__'):
-            return DataCollection(unary_op.__dataframe_apply__(self._iterable))
-
-        return self._factory(filter(inner, self._iterable))
+            child = self._factory(self._iterable.filter(unary_op))
```
[the remark this answers is not in the document]
Comment (zhujiming) [reply towhee/functional/data_collection.py:419]: code style

**Review by reiase — APPROVED** [review]

**Pull request merged**